### PR TITLE
[builder] CxG schema 5 / Census schema 2

### PR DESCRIPTION
**Pull request by bkmartinjr**
Builder support for Census schema 2.0.0 / CELLxGENE schema 5.0.0

Fixes: #993 
Fixes: #1022 
Fixes: #796

Primary changes:
* Request H5ADs in schema 5.0.0 format
* Update Census schema to 2.0.0, in turn is pinned to CxG schema 5.0.0
* Update assay filter to incorporate new RNA seq assays
* Update UBERON release to match schema 5.0.0, used to infer tissue_general value
* Clarify the handling of multi-species datasets and how they are included/excluded from the Census. As part of this, tighten typing on ExperimentSpecification to require a species/organism to be specified (i.e., mandate that each Experiment is associated with a single organism).
* Add new census_info/organisms table specified in Census schema 2.0.0 (#796)

### CODE REVIEW

**Comment by bkmartinjr** [comment]
@pablo-gar - thanks for the assay term changes.

Question/open issue:  the Census schema also defines special handling for the normalized layer, for any "Smart-Seq" assays.  Can we add the definitive list of which EFO terms are considered a Smart-Seq assay?  Ideally this would be included in the Census schema (or referenced as you did with the assay filter terms).

**Review by ebezzi — APPROVED** [review]
LGTM.

File: docs/cellxgene_census_schema.md [hunk 1]
```diff
@@ -339,15 +223,15 @@ An example of this `SOMADataFrame` is shown below:
 <tbody>
   <tr>
     <td>census_schema_version</td>
-    <td>1.3.0</td>
+    <td>2.0.0</td>
   </tr>
   <tr>
     <td>census_build_date</td>
     <td>2022-11-30</td>
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply docs/cellxgene_census_schema.md]: fixed

**Comment by prathapsridharan** [comment]
@bkmartinjr - In regards to issue #993  there is this point:

```
Confirm no use of CL:0000003, aka native cell. These cells will now be marked as unknown in both obs.cell_type and obs.cell_type_ontology_term_id columns. See related task: https://github.com/chanzuckerberg/cellxgene-census/issues/1019
```

Once a test build is produced, are we to check this _manually in the python interpreter_ by doing a count of `CL:0000003` in `obs.cell_type_ontology_term_id` and `native cell ` in `obs.cell_type` and checking that the count is zero? 

Or should this be captured in some type of of _post build_ acceptance test where some sanity checks about the data are done?

**Comment by bkmartinjr** [comment]
@prathapsridharan re:
>  are we to check

This is entirely an upstream issue in the DP process, and the builder does not enforce (or check) for this level of metadata compliance with the CxG schema.  These checks for compliance with the CxG schema are the provenance of the schema validation toolkit used by Lattice, et al.

We _could_ do this kind of checks, but it is redundant, and adds linkages across layers in the system that don't add much value IMHO.

**Review by prathapsridharan — APPROVED** [review]
LGTM

**Review by pablo-gar — APPROVED** [review]
I took a look at the schema updates and the updated list of accepted assays. Looks good!

**Comment by bkmartinjr** [comment]
@atarashansky - we have included your requested organisms info (#796) in Census schema 2.0.0.  Current content will be:
```
In [4]: census['census_info']['organisms'].read().concat().to_pandas().set_index('soma_joinid')
Out[4]: 
            organism_ontology_term_id organism_label      organism
soma_joinid                                                       
0                      NCBITaxon:9606   Homo sapiens  homo_sapiens
1                     NCBITaxon:10090   Mus musculus  mus_musculus
```

Please feel free to leave comments on both the Schema MD file changes and the code.

**Comment by pablo-gar** [comment]
Light QC on test build shows no issues

Checks
- Verifying fidelity in data additions, in particular changes to `obs` and addition to `census["census_info"]["organisms"]`.
- Checking new assays added based on updated list of assays, and verifying the addition of expected datasets.
- Validating the existence of  previously missing data due to fixed filter for multi-species.
- Validating calculation of normalized layer for SMART-like technologies

See notebook
https://colab.research.google.com/drive/1kb2ZR0MPxVsWBgJlIgrtJrkGPx5wqxIk

**Pull request merged**